src/io/sqlite_store.rs

```diff
@@ -0,0 +1,228 @@
+use super::*;
+
+use lightning::util::persist::KVStorePersister;
+use lightning::util::ser::Writeable;
+
+use rusqlite::{named_params, Connection};
+
+use std::fs;
+use std::io::Cursor;
+use std::path::PathBuf;
+use std::str::FromStr;
+use std::sync::{Arc, Mutex};
+
+// The database file name.
+const SQLITE_DB_FILE: &str = "ldk_node.sqlite";
+
+// The table in which we store all data.
+const KV_TABLE_NAME: &str = "ldk_node_data";
+
+// The current SQLite `user_version`, which we can use if we'd ever need to do a schema migration.
+const SCHEMA_USER_VERSION: u16 = 1;
+
+/// A [`KVStore`] implementation that writes to and reads from an [SQLite] database.
+///
+/// [SQLite]: https://sqlite.org
+pub struct SqliteStore {
+	connection: Arc<Mutex<Connection>>,
+}
+
+impl SqliteStore {
+	pub(crate) fn new(dest_dir: PathBuf) -> Self {
```

Comment:
Should this return an error so that any failure doesn't cause a mobile app to crash? I'm assuming that's what `expect` will do, though I'm not sure if there is any way to catch this in other languages.